### PR TITLE
ras/smtstate.py: Added new test

### PR DESCRIPTION
**Pull request by rohan-r-deshpande**
- Added a new test "test_smt_state_switching" to validate the time taken for switching between different SMT levels
- Added a yaml file: smtstate.py.data/smtstate.yaml

Signed-off-by: Rohan Deshpande <rohan_d@linux.ibm.com>

### CODE REVIEW

File: ras/smtstate.py [hunk 1]
```diff
@@ -67,3 +73,26 @@ def test(self):
...
+        """
+        Test to check the time taken for switching between different
+        SMT levels. The time taken must not exceed beyond 5 minutes.
+        """
+        process.system("ppc64_cpu --smt=on")
+
+        for i in [0, 1, 2, 4]:
+            if i == 0:
+                cmd_output = process.run(
+                    "/usr/bin/time -p ppc64_cpu --smt off", shell=True)
```
Comment (PraveenPenguin) [inline ras/smtstate.py:87]: is `/usr/bin/time` is available for no root user if not i recommend please use `sudo=True`

File: ras/smtstate.py [hunk 2]
```diff
@@ -12,6 +12,11 @@
 # See LICENSE for more details.
 # Copyright: 2021 IBM
 # Author: Pavithra Prakash <pavrampu@linux.vnet.ibm.com>
+#
+# Modified: 22/11/2022
+# Author: Rohan Deshpande <rohan_d@linux.vnet.ibm.com>
+# Description: Added new test "test_smt_state_switching"
```
Comment (PraveenPenguin) [inline ras/smtstate.py:18]: i feel not require this change as git history contains same

File: ras/smtstate.py [hunk 3]
```diff
@@ -67,3 +73,26 @@ def test(self):
                     self.log.info("SMT load is successful for SMT=%s" % j)
                 else:
                     self.fail("smt load failed")
+
+    def test_smt_state_switching(self):
+        """
+        Test to check the time taken for switching between different
+        SMT levels. The time taken must not exceed beyond 5 minutes.
```
Comment (PraveenPenguin) [inline ras/smtstate.py:80]: not sure if this statement holds good and may be reworded like 

SMT levels. The time taken in general cases should not exceed 5 minutes.

File: ras/smtstate.py [hunk 4]
```diff
@@ -67,3 +73,26 @@ def test(self):
...
+                    "/usr/bin/time -p ppc64_cpu --smt off", shell=True)
+            elif i == 1:
+                cmd_output = process.run(
+                    "/usr/bin/time -p ppc64_cpu --smt on", shell=True)
+            else:
+                cmd_output = process.run(
+                    "/usr/bin/time -p ppc64_cpu --smt %s", i, shell=True)
+
+            if (int(cmd_output.duration) > self.time_in_minutes):
+                self.fail("ERROR : SMT has taken longer than expected")
```
Comment (PraveenPenguin) [inline ras/smtstate.py:96]: here also please FAIL: SMT has taken longer than expected

File: ras/smtstate.py (hunk 1 above)
Comment (rohan-r-deshpande) [reply ras/smtstate.py:87]: I suppose, it is available, but _sudo=True_ better approach, just in case.

File: ras/smtstate.py (hunk 2 above)
Comment (rohan-r-deshpande) [reply ras/smtstate.py:18]: Sure, will get it removed.

File: ras/smtstate.py (hunk 3 above)
Comment (rohan-r-deshpande) [reply ras/smtstate.py:80]: Noted.

File: ras/smtstate.py (hunk 4 above)
Comment (rohan-r-deshpande) [reply ras/smtstate.py:96]: Fine. Will change it!

File: ras/smtstate.py (hunk 1 above)
Comment (rohan-r-deshpande) [reply ras/smtstate.py]: Done

File: ras/smtstate.py (hunk 2 above)
Comment (rohan-r-deshpande) [reply ras/smtstate.py]: Done

File: ras/smtstate.py (hunk 3 above)
Comment (rohan-r-deshpande) [reply ras/smtstate.py]: Done

File: ras/smtstate.py (hunk 4 above)
Comment (rohan-r-deshpande) [reply ras/smtstate.py]: Done

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**